### DISCUSSION TITLE
Do I need to keep resetting file permissions in local dev?

### DISCUSSION THREAD

**Discussion by KarlBishop** [craftcms/cms#13875] [Q&A] [answered]
When I started local dev-ing with Nitro yonks ago, part of the instruction was to [set various file permissions to 777](https://craftcms.com/docs/nitro/2.x/windows.html#set-permissions-when-adding-a-new-site).

I often needed to reset these file permissions e.g. when installing plugins, updating Craft or changing anything that touches the project config. And I found a few other folders needed to be 777ed occasionally, so I came to frequently running this command:

````
sudo chmod -R 777 ./{.env,composer.json,composer.lock,.git,config,modules,storage,templates,vendor,web}
````

Now this is starting to be a pain. And I noticed there doesn't seem to be an instruction about file permissions in the [latest install guide](https://craftcms.com/docs/getting-started-tutorial/install/).

So I'm wondering:

- Are 777 file permissions no longer needed when using DDEV?
- If they are still needed, is there a way to auto-reapply them after every install, update or change?

**Comment by thupsi** [comment, marked as answer]
Normaly with DDEV you won't need to mess with file permissions at all. 

It all works as it should from the start and doesn't break after install, update or change.
